projects/lexical_shared/lib/lexical/document.ex

```diff
@@ -43,12 +44,27 @@ defmodule Lexical.Document do
   @spec new(Lexical.path() | Lexical.uri(), String.t(), version()) :: t
   def new(maybe_uri, text, version) do
     uri = DocumentPath.ensure_uri(maybe_uri)
+    path = DocumentPath.from_uri(uri)
+    lang_id = DocumentPath.language_id_from_path(path)
 
     %__MODULE__{
       uri: uri,
       version: version,
       lines: Lines.new(text),
-      path: DocumentPath.from_uri(uri)
+      path: path,
+      language_id: lang_id
```

Comment:
suggestion:
```elixir
path = DocumentPath.from_uri(uri)
language_id = uri |> DocumentPath.from_uri() |> language_id_from_path()
new(maybe_uri, text, language_id, version)
```

Reply:
I've updated the function for creating a new document, positioning `language_id` as an optional argument. lmk what you think.

Reply:
[Moving forward w/ this here](https://github.com/lexical-lsp/lexical/pull/662#discussion_r1541386976).